client/tests/test_class_api.py

```diff
@@ -0,0 +1,222 @@
+    original_syntax_state = list(
+        filter(
+            lambda r: isinstance(r, (Entity, Feature, Label)), ff.state().sorted_list()
+        )
+    )
+
+    ff.clear_state()
+
+    arrange_resources(
+        provider, source, inference_store, is_local, is_insecure, is_class_api=True
+    )
+    class_syntax_state = list(
+        filter(
+            lambda r: isinstance(r, (Entity, Feature, Label)), ff.state().sorted_list()
+        )
+    )
+
+    assert original_syntax_state == class_syntax_state
+
+
+@pytest.mark.parametrize(
+    "provider_source_fxt,is_local",
+    [
+        pytest.param("local_provider_source", True, marks=pytest.mark.local),
+        pytest.param(
+            "hosted_sql_provider_and_source", False, marks=pytest.mark.hosted
+        ),
+        pytest.param(
+            "hosted_sql_provider_and_source", False, marks=pytest.mark.docker
+        ),
+    ],
+)
+def test_variants_naming_consistency(
+    provider_source_fxt, is_local, request
+):
+    custom_marks = [
+        mark.name for mark in request.node.own_markers if mark.name != "parametrize"
+    ]
+    _, source, _ = request.getfixturevalue(provider_source_fxt)(
+        custom_marks
+    )
+    label_column = "IsFraud" if is_local else "isfraud"
+    label_entity_column = "CustomerID" if is_local else "customerid"
+
+    with pytest.raises(ValueError):
+        ff.Variants(
+            {
+                "quickstart": ff.Label(
+                    source[[label_entity_column, label_column]],
+                    description="Whether a user's transaction is fraudulent.",
+                    variant="quickstart_v2",
+                    type=ff.Bool,
+                ),
+                "quickstart_v2": ff.Label(
+                    source[[label_entity_column, label_column]],
+                    description="Whether a user's transaction is fraudulent.",
+                    variant="quickstart",
+                    type=ff.Bool,
+                ),
+            }
+        )
+
+
+@pytest.mark.parametrize(
+    "provider_source_fxt,is_local",
+    [
+        pytest.param("local_provider_source", True, marks=pytest.mark.local),
+        pytest.param(
+            "hosted_sql_provider_and_source", False, marks=pytest.mark.hosted
+        ),
+        pytest.param(
+            "hosted_sql_provider_and_source", False, marks=pytest.mark.docker
+        ),
+    ],
+)
+def test_indexing_with_fewer_than_two_columns(
+    provider_source_fxt, is_local, request
+):
+    custom_marks = [
+        mark.name for mark in request.node.own_markers if mark.name != "parametrize"
+    ]
+    _, source, _ = request.getfixturevalue(provider_source_fxt)(
+        custom_marks
+    )
+    label_entity_column = "CustomerID" if is_local else "customerid"
+
+    with pytest.raises(Exception, match="Expected 2 columns"):
+        source[[label_entity_column]]
+
+
+@pytest.fixture(autouse=True)
+def before_and_after_each(setup_teardown):
```

Comment:
this has been copied pasta'd in a lot of places. I'm not sure what the correct solution is but ideally we don't' have to.